src/librustc_typeck/check/method/probe.rs

```diff
@@ -43,6 +43,7 @@ struct ProbeContext<'a, 'tcx:'a> {
     extension_candidates: Vec<Candidate<'tcx>>,
     impl_dups: HashSet<ast::DefId>,
     static_candidates: Vec<CandidateSource>,
+    predicates: Vec<TraitRef<'tcx>>
```

Comment:
Nit: let's give this a name like "unsatisfied_candidates". It seems like it'd be good to add a comment (perhaps above `static_candidates`) explaining that these two fields collect "near misses" for use in later error-reporting and don't play a role in method selection per se.